ToolBox/BitfinexDownloader/BitfinexExchangeInfoDownloader.cs

```diff
@@ -0,0 +1,156 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Newtonsoft.Json.Linq;
+using QuantConnect.Logging;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.ToolBox.BitfinexDownloader
+{
+    /// <summary>
+    /// Bitfinex implementation of <see cref="IExchangeInfoDownloader"/>
+    /// </summary>
+    public class BitfinexExchangeInfoDownloader : IExchangeInfoDownloader
+    {
+        /// <summary>
+        /// Currency mapping
+        /// </summary>
+        private readonly Dictionary<string, string> _oldWithNewCurrencies = new()
+        {
+            { "UST", "USDT" },
+            { "ALG", "ALGO" },
+            { "AMP", "AMPL" },
+            { "ATO", "ATOM" },
+            { "B21X", "B21" },
+            { "XCH", "XCHF" },
+            { "DAT", "DATA" },
+            { "DOG", "MDOG" },
+            { "DSH", "DASH" },
+            { "ETH2X", "ETH2" },
+            { "EUS", "EURS" },
+            { "EUT", "EURT" },
+            { "GNT", "GLM" },
+            { "IDX", "ID" },
+            { "IOT", "IOTA" },
+            { "PAS", "PASS" },
+            { "QTM", "QTUM" },
+            { "RBT", "RBTC" },
+            { "REP", "REP2" },
+            { "STJ", "STORJ" },
+            { "TSD", "TUSD" },
+            { "UDC", "USDC" },
+            { "WBT", "WBTC" },
+            { "OMN", "OMNI" },
+            { "MNA", "MANA" },
+        };
+
+        /// <summary>
+        /// Market name
+        /// </summary>
+        public string Market => QuantConnect.Market.Bitfinex;
+
+        /// <summary>
+        /// Pulling data from a remote source
+        /// </summary>
+        /// <returns>Enumerable of exchange info</returns>
+        public IEnumerable<string> Get()
+        {
+            const string tradingPairsUrl = "https://api-pub.bitfinex.com/v2/conf/pub:list:pair:exchange";
+            const string currenciesUrl = "https://api-pub.bitfinex.com/v2/conf/pub:list:currency";
+            const string pairInfosUrl = "https://api-pub.bitfinex.com/v2/conf/pub:info:pair";
+            const string pairLabelUrl = "https://api-pub.bitfinex.com/v2/conf/pub:map:currency:label";
+            Dictionary<string, string> headers = new() { { "User-Agent", ".NET Framework Test Client" } };
```

Comment:
> .NET Framework Test Client

Nit - we are not really using net framework/core but it's just called .Net now I believe. Same for GDAX 👍